hawkmoth/__init__.py

```diff
@@ -8,6 +8,7 @@
 """
 
 import glob
+import logging as log
```

Comment:
So far I'm enjoying these patches, but I have serious doubts about this one. As far as I can see, you don't save nearly enough verbosity to justify the obfuscation of renaming one of the core modules. So I'd rather do `from sphinx.util import logging as log` all other considerations aside.

However, I think using the levels directly from `logging` is also wrong technically, even if it happens to produce the right outcome _today_. Note that `sphinx.util.logging` is clearly based on Python's `logging` (we can even find it inside under `sphinx.util.logging.logging`), and `sphinx.util.logging.LEVEL_NAMES` clearly map those of Python's `logging`. Those are still implementation details, yet further down we get this:

```python
    logger = logging.getLogger(__name__)
             ^^^^^^^
    [...]
    if self.__parser_verbosity(error.level) <= self.env.app.verbosity:
        self.logger.log(error.level, error.get_message(), location=(self.env.docname, self.lineno))
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

So, if I followed the code correctly, we're using the `sphinx.util` logger (naturally), but specifying the log level with Python's `logging` constants. At the very least, I'd use `sphinx.util.logging.logging.ERROR` and so on, but even that is inferior to using `sphinx.util.logging.LEVEL_NAMES` in my view as nothing prevents Sphinx from registering its own log levels.

Reply:
> As far as I can see, you don't save nearly enough verbosity to justify the obfuscation of renaming one of the core modules. So I'd rather do `from sphinx.util import logging as log` all other considerations aside.

This was just to avoid a name collision with the Sphinx package, not to make it shorter, really. I only needed it for `log.ERROR` and `log.WARNING`. But I agree it's less than stellar.

Reply:
With the rest, the part I struggle with is figuring out what is Sphinx API and what is implementation details. Are the log level values API? Are the log level names API? Is the `sphinx.util.logging.LEVEL_NAMES` dictionary API?

The API documentation https://www.sphinx-doc.org/en/master/extdev/logging.html doesn't really say anything about any of them.

The `SphinxLoggerAdapter` implementation does not override the `LoggerAdapter` `info()`, `warning()`, etc. methods, and based on the Python documentation https://docs.python.org/3/library/logging.html#loggeradapter-objects they use the corresponding Python `logging` log levels. Sure, Sphinx could override them and use different levels, but to me it seems far fetched.

Reply:
> This was just to avoid a name collision with the Sphinx package [...]

I got that. Just bear in mind you can alias the other one instead and keep the standard one with its original name, which might be preferable.

> [...] Sure, Sphinx could override them and use different levels,  but to me it seems far fetched.

Yeah, I would also bet this will never be a problem, and yet it feels kind of dirty to do it this way. Maybe doing `logging.logging.ERROR` is an acceptable compromise? Or maybe make two imports like so:

```python
from sphinx.util import logging as log
from sphinx.util.logging import logging
```

It would at least remove the blind assumption that Sphinx logger is based on the standard Python logger.

Reply:
I wanted to remove the `ErrorLevel` enum because I thought it was redundant, and forced us to map Clang diagnostics to `ErrorLevel` and `ErrorLevel` to Sphinx error levels, i.e. two maps.

Since I don't want the parser to know anything about Sphinx, nor do I want the extension to know (much) about Clang, I think it stands to reason to use Python logging error levels in the parser. But based on what you're saying, we still need to map those to Sphinx error levels in the extension. Even though for all practical purposes they match.

I think I'll need to give this some more thought.

Reply:
Added new pull request #98 which I hope addresses all the concerns here. Fingers crossed. ;)

Reply:
As always, the review comments are much appreciated!